src/opendr/perception/pose_estimation/hr_pose_estimation/HighResolutionLearner.py

```diff
@@ -0,0 +1,992 @@
+        start.record()
+
+        stages_output = net(tensor_img)
+
+        end.record()
+        torch.cuda.synchronize()
+
+        stage2_pafs = stages_output[-1]
+        pafs = np.transpose(stage2_pafs.squeeze().cpu().data.numpy(), (1, 2, 0))
+
+        avg_pafs = pafs
+        return avg_pafs
+
+    def second_pass_infer(self, net, img, net_input_height_size, max_width, stride, upsample_ratio,
+                          pad_value=(0, 0, 0),
+                          img_mean=np.array([128, 128, 128], np.float32), img_scale=np.float32(1 / 256)):
+        height, width, _ = img.shape
+        scale = net_input_height_size / height
+        img_ratio = width / height
+        if img_ratio > 6:
+            scale = max_width / width
+
+        scaled_img = cv2.resize(img, (0, 0), fx=scale, fy=scale, interpolation=cv2.INTER_LINEAR)
+        scaled_img = normalize(scaled_img, img_mean, img_scale)
+        min_dims = [net_input_height_size, max(scaled_img.shape[1], net_input_height_size)]
+        padded_img, pad = pad_width(scaled_img, stride, pad_value, min_dims)
+
+        if 'cuda' in self.device:
+            tensor_img = torch.from_numpy(padded_img).permute(2, 0, 1).unsqueeze(0).float().cuda()
+            tensor_img = tensor_img.cuda()
+            if self.half:
+                tensor_img = tensor_img.half()
+        else:
+            tensor_img = torch.from_numpy(padded_img).permute(2, 0, 1).unsqueeze(0).float().cpu()
+
+        start = torch.cuda.Event(enable_timing=True)
+        end = torch.cuda.Event(enable_timing=True)
+        start.record()
+
+        stages_output = net(tensor_img)
+
+        end.record()
+        torch.cuda.synchronize()
+
+        stage2_heatmaps = stages_output[-2]
+        heatmaps = np.transpose(stage2_heatmaps.squeeze().cpu().data.numpy(), (1, 2, 0))
+        heatmaps = heatmaps.astype(np.float32)
+        heatmaps = cv2.resize(heatmaps, (0, 0), fx=upsample_ratio, fy=upsample_ratio, interpolation=cv2.INTER_CUBIC)
+
+        stage2_pafs = stages_output[-1]
+        pafs = np.transpose(stage2_pafs.squeeze().cpu().data.numpy(), (1, 2, 0))
+        pafs = pafs.astype(np.float32)
+        pafs = cv2.resize(pafs, (0, 0), fx=upsample_ratio, fy=upsample_ratio, interpolation=cv2.INTER_CUBIC)
+
+        return heatmaps, pafs, scale, pad
+
+    def Pooling(self, img, kernel):  # Pooling on input image for dim reduction
+        pooling = torch.nn.AvgPool2d(kernel)
+        pool_img = torchvision.transforms.ToTensor()(img)
+        pool_img = pool_img.unsqueeze(0)
+        pool_img = pooling(pool_img)
+        pool_img = pool_img.squeeze(0).permute(1, 2, 0).cpu().float().numpy()
+        return pool_img
+
+    def fit(self, val_dataset=None, logging_path='', logging_flush_secs=30,
+            silent=False, verbose=True, use_val_subset=True, val_subset_size=250,
+            val_images_folder_name="val2017", val_annotations_filename="person_keypoints_val2017.json"):
```

Comment:
```suggestion
    def fit(self, dataset, val_dataset=None, logging_path='', silent=True, verbose=True):
```

Reply:
Should match signature of parent method.

Reply:
changed